internal/verifier/web_server.go

```diff
@@ -177,15 +185,38 @@ func (server *WebServer) Run(ctx context.Context) error {
 // EmptyRequest is for request with empty body
 type EmptyRequest struct{}
 
+// CheckRequest is for requests to the /check endpoint.
+type CheckRequest struct {
+	Filter map[string]any `json:"filter"`
+}
+
 func (server *WebServer) checkEndPoint(c *gin.Context) {
-	var json EmptyRequest
+	var req CheckRequest
 
-	if err := c.ShouldBindJSON(&json); err != nil {
+	if err := c.ShouldBindJSON(&req); err != nil {
+		err = errors.Wrap(err, "filter is not valid JSON")
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
 
-	server.Mapi.Check(context.Background())
+	var filter bson.D
+	if req.Filter != nil {
+		raw, err := bson.Marshal(req.Filter)
+		if err != nil {
+			err = errors.Wrap(err, "failed to marshal json to bson.Raw")
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
```

Comment:
Wouldn’t an error here be an internal server error (HTTP 500)? “Bad Request” would indicate that there’s something the user could do to change the outcome, but if all we’re doing here is turning a map into BSON bytes, that seems like something the user can’t influence; i.e., it’s internal to the server.

Reply:
Now that we're using `map[string]any` directly for `filter`, this error does not occur in the web server.